cloud/apitoken/apitoken.go

```diff
@@ -0,0 +1,257 @@
+package apitoken
+
+import (
+	httpContext "context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	astrocore "github.com/astronomer/astro-cli/astro-client-core"
+	"github.com/astronomer/astro-cli/context"
+	"github.com/astronomer/astro-cli/pkg/ansi"
+	"github.com/astronomer/astro-cli/pkg/input"
+	"github.com/astronomer/astro-cli/pkg/printutil"
+)
+
+var (
+	ErrInvalidAPITokenKey           = errors.New("invalid ApiToken selected")
+	ErrNoAPITokenNameProvided       = errors.New("you must give your ApiToken a name")
+	ErrNoAPITokensFoundInDeployment = errors.New("no ApiTokens found in your deployment")
+	apiTokenPagnationLimit          = 100
+)
+
+func CreateDeploymentAPIToken(name, role, description, deployment string, out io.Writer, client astrocore.CoreClient) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+
+	if name == "" {
+		fmt.Println("Please specify a name for your ApiToken")
+		name = input.Text(ansi.Bold("\nApiToken name: "))
+		if name == "" {
+			return ErrNoAPITokenNameProvided
+		}
+	}
+
+	mutateAPITokenInput := astrocore.CreateDeploymentApiTokenRequest{
+		Role:        role,
+		Name:        name,
+		Description: &description,
+	}
+	resp, err := client.CreateDeploymentApiTokenWithResponse(httpContext.Background(), ctx.Organization, deployment, mutateAPITokenInput)
```

Comment:
we should save the api token from response to ASTRO_API_TOKEN variable. Right now, the token gets created, but user cannot use the token value

Reply:
will get a pr up for your comments

Reply:
I think instead of setting it we really want to print it to screen so they can use it where they want to.

Reply:
worried about the token leaking in ci cd logs...